test/promql/tests/test_notready_pods.yaml

```diff
@@ -0,0 +1,162 @@
+    input_series:
+      # Pod is in the Failed state and scheduled, so it will be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Pending"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Unknown"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Failed"}'
+        values: '1x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Running"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Succeeded"}'
+        values: '0x14'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-3", pod="pod-1"}'
+        values: '0x14'
+
+    alert_rule_test:
+      - eval_time: 15m
+        alertname: PodsNotReady
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: ns-3
+              pod: pod-1
+              phase: Failed
+            exp_annotations:
+              summary: "Pod pod-1 is not ready"
+              description: "Pod pod-1 in namespace ns-3 is not ready for more than 15 minutes."
+
+  - interval: 1m
+    input_series:
+      # Not Alerted cases:
+      # Pod is in the Unknown state and unscheduled (unscheduled once in the 15 min interval), so it will not be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Pending"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Unknown"}'
+        values: '1x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Failed"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Running"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Succeeded"}'
+        values: '0x14'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-4", pod="pod-1"}'
+        values: '1x0 0x13'
+
+      # Pod is in Succeeded state and scheduled, so it will not be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Pending"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Unknown"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Failed"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Running"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Succeeded"}'
+        values: '1x14'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-5", pod="pod-1"}'
+        values: '0x14'
+
+      # For the first 5 minutes, Pod is in the Pending state, next 5 minutes it's in the Unknown state, for last 5 minutes it's in the Failed state and scheduled, so it will be not alerted.
```

Comment:
I'd expect it to be alerting. Do you know why it is not alerting? Judging by the rule itself, it should be alerting. Might be that there's some sort of edge case here.

Reply:
Maybe try with something like this to cover some sort of edge case with the transition between the stages:
```
      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-6", phase="Pending"}'
        values: '1x5 0x3 0x4'
      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-6", phase="Unknown"}'
        values: '0x4 1x5 0x3'
      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-6", phase="Failed"}'
        values: '0x4 0x4 1x4'
```
I would not expect that to be an issue, but I think this sort of setup should rule that out at least.